src/vs/workbench/contrib/terminal/browser/xterm/cognisantCommandTrackerAddon.ts

```diff
@@ -100,26 +105,31 @@ export class CognisantCommandTrackerAddon extends CommandTrackerAddon {
 					}
 				}
 				break;
-			case ShellIntegrationInteraction.CommandFinished:
+			case ShellIntegrationInteraction.CommandFinished: {
 				this._logService.trace('Terminal Command Finished', this._currentCommand.command);
 				this._exitCode = Number.parseInt(event.value);
 				if (!this._currentCommand.marker?.line || !this._terminal.buffer.active) {
 					break;
 				}
+				let output = '';
+				for (let i = this._currentCommand.marker.line + 1; i <= this._terminal.buffer.active.cursorY; i++) {
+					output += this._terminal.buffer.active.getLine(i)?.translateToString() + '\n';
```

Comment:
This will be too much memory to duplicate (and too much work to do for every command when it will rarely be used), we should do this only when requested using markers and if the output has been trimmed off the top then disable the option.